### PR TITLE
Set sensitive values as secret - Part 3

### PR DESCRIPTION
**Pull request by taylor-swanson**
## Proposed commit message

- Set sensitive values as secret in integrations.
- Updated package-spec to 3.0.2 in most packages to leverage secrets validation checks.
- Certain packages required fixes to mappings (due to missing or incorrect mappings).
- Certain packages were not able to go to package-spec 3.0.2 due issues with dynamic mappings. These have been held back at 3.0.0.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates elastic/security-team#7388

### CODE REVIEW

File: packages/o365/data_stream/audit/fields/fields.yml [hunk 1]
```diff
@@ -7,7 +7,7 @@
         - name: ID
           type: keyword
         - name: Type
-          type: keyword
+          type: long
```
Comment (kcreddy) [inline packages/o365/data_stream/audit/fields/fields.yml:10]: `Actor.Type` and `Target.Type` are both mapped to[ same type](https://learn.microsoft.com/en-us/office/office-365-management-api/office-365-management-activity-api-schema#complex-type-identitytypevaluepair). Maybe we could make it both `integer` or `long`.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

File: packages/o365/data_stream/audit/fields/fields.yml (hunk 1 above)
Comment (taylor-swanson) [reply packages/o365/data_stream/audit/fields/fields.yml:10]: I think I'll actually drop in a convert processor here. Dan [mentioned this here](https://github.com/elastic/integrations/pull/8725#discussion_r1486899571) and I think it's applicable in this case as well. In general, even though it is technically a number, what we really want is it indexed as is a keyword.

**Review by kcreddy — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**